### DISCUSSION TITLE
Client API Codegen for kotlin to generated Javacode Issue

### DISCUSSION THREAD

**Discussion by parkinyoung** [Netflix/dgs-framework#467] [Q&A]
When I was update "com.netflix.dgs.codegen" version to "5.0.2" and generatedClient option change to true for JUnit testing in kotlin project

I found Client API code generated to java code. and I saw https://github.com/Netflix/dgs-codegen/pull/166 PR

In my project, using scalar Long type(with graphql-dgs-extended-scalars) and add typeMapping "Long" to "kotlin.Long"
When using version 4.6.x, it working well. Because generated code type is kotlin.

If add typeMapping "Long" to "java.lang.Long", All kotlin data class types that have Long type are changed to java.long
What can I do?  Is it only way to divide module?

**Comment by paulbakker** [comment]
I don't think you need to add a `typeMapping` for long. I just tried it, and it seems to work correctly without.

**Reply by parkinyoung** [reply]
First, Thank you for quick response :)
The capture below is the generated code without specifying the type.
(The package name is arbitrarily changed for sharing.)
![image](https://user-images.githubusercontent.com/8326293/124646915-380ef400-ded0-11eb-8249-c650a50fe127.png)
As you can see from the above capture, the generated code tries to find a class called "Long" under the default package "types".

I want to use Long type as Input parameter. (When the corresponding type is not used as a parameter, it works well )
In this case, Is there no need to specify the type?

Additionally, I thick that no handling for Long types in`TypeName.toJavaTypeName()`. so I guess it will return `ClassName.get(packageName, simpleName)` 

Thanks again for the test and answer
